students/cbrown/Lesson 9/mailroom_oo.py

```diff
@@ -0,0 +1,223 @@
+    '''
+    thanks_dict = {'donor_name': donor.name, 'amount': donor.amount}
+
+    # Formatted Thank You Note
+    message = ('Dear {donor_name}, \n\nThank you for your show of support and generosity. '
+      'Your Donation of ${amount} will contribute to saving Olympic Marmots '
+      'in Washington State. These Marmota are special and a unique gift to the Olympic '
+      'National Park ecosystem. As a way of saying thank you. '
+      'You will be receiving your very own Olympic Marmot t-shirt in the mail!\n\n'
+      'Sincerely,\n\nThe Olympic Marmot Wildlife Foundation\n').format(**thanks_dict)
+
+    return message
+
+
+def send_thanks():
+    '''
+    Adds new donations and new donors. Prints out message to individual donors.
+    '''
+    print('\n', "For A Complete Donor List Type 'List'\n ")
+
+    #Displays the list of current donors for the user if requested
+    while True:
+        donor_name = input("What donor(s) are you looking for?\n ").title()
+        if donor_name == 'List':
+            print('\n', "Here is A Complete List of Donor Names\n ")
+            donor_list = [donor for donor in donor_collection.donor_dict.keys()]
+            print(donor_list)
+        else:
+            break
+
+    # prompt for donation amount
+    try:
+        amount = float(input("How Much Did This Person Donate?\n "))
+
+    except ValueError:  #start prompt over again if exception occurs
+        print('Please input a numeric amount & try again.')
+        send_thanks()
+
+    #Creates new instance of donor
+    donor = Donor(donor_name, amount)
+
+    # updates donor database & adds new ones if a name does not exist
+    donor_verify(donor)
+
+    # formats message thanking the individual donor
+    message = thank_you_note(donor)
+
+    print(message)
+
+
+def create_message(donor):
+    '''
+    Creates a personalized & formatted message for each donor
+    '''
+    tmt_amount = donor.sum_amount()
+
+    message = ('Dear {}, \n\nThank you for your show of support and generosity. '
+        'Your donations totaling ${:,.1f} will contribute to saving Olympic Marmots '
+        'in Washington State. These Marmota are special and a unique gift to the Olympic '
+        'National Park ecosystem. As a way of saying thank you. '
+        'You will be receiving your very own Olympic Marmot t-shirt in the mail!\n\n'
+        'Sincerely,\n\nThe Olympic Marmot Wildlife Foundation\n').format(donor.name, tmt_amount)
+
+    return message
+
+
+def write_file(name, message):
+    '''
+    Creates a text file of each thank you message for each donor
+    '''
+    fname = name + '.txt'
+    with open(fname, 'w') as f:
+        f.write(message)
+
+
+def send_letters():
+    '''
+    creates and writes a letter to each donor in the donor database
+    '''
+
+    for name, amount in donor_collection.donor_dict.items():
+        donor = Donor(name, amount)
+
+        message = create_message(donor)
+
+        write_file(donor.name, message)
```

Comment:
your cli code should be interacting with DonorCollection class only. 

Donor objects are referenced in the DonorCollection class.

see example:
https://repl.it/@naaleksandrova/Lesson9Example#main.py